### PR TITLE
Adding USERS directory to create a list of Cilium users

### PR DESCRIPTION
**Pull request by tgraf**
Sharing experiences and learning from other users is important. We are
frequently asked who is using a particular feature of Cilium in order to get in
contact with other users of a particular features to share experiences and
best-practices. While the Slack community allows users to get in touch, it can
be challenging to quickly find users of a particular feature.

---
This change is [Reviewable](https://reviewable.io/reviews/cilium/cilium/9810)

### CODE REVIEW

**Comment by tgraf** [comment]
test-docs-please

**Comment by tgraf** [comment]
test-docs-please

**Review by aanm — APPROVED** [review]

File: USERS.md [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+   deployments only, i.e., no demo or trial deployments. Commercial or
+   production use is not required. A well-done home lab setup can be equally
+   interesting as a large-scale commercial deployment.
+
+Users (Alphabetically)
+---------------------
+
+    * N: Cilium Example User Inc.
+      D: Cilium Example User Inc. is using Cilium for scientific purposes
+      U: ENI networking, DNS policies, ClusteMesh
```
Comment (brb) [inline USERS.md:41]: Nit: too late, but `s/ClusteMesh/ClusterMesch/`.

File: USERS.md [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+
+If you are using Cilium, please consider adding yourself as a user with a quick
+description of your use case by opening a pull request to this file and adding
+a section describing your usage of Cilium. If you are open to others contacting
+you about your use of Cilium on Slack, add your Slack nick as well.
+
+    N: Name of user (company or individual)
+    D: Description
+    U: Usage of features
+    Q: Contacts available for questions
```
Comment (borkmann) [inline USERS.md:24]: I presume adding information about the scale of the deployment might deprecate too quickly?

**Pull request merged**